### PR TITLE
Add lazy_import

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

_Created with Grayskull_

### CODE REVIEW

**Comment by BastianZim** [comment]
I'm not completely sure about the license. The package is based on another one and lists both, which is why I included both licenses, but I don't know if that's the correct approach, since the linter already complained (Without the second one, it works). 

See: https://github.com/mnmelo/lazy_import#license

Pinging @conda-forge/staged-recipes

**Comment by BastianZim** [comment]
`conda smithy recipe-lint .` output:
```zsh
(conda-smithy) bastianzimmermann@Bastians-iPro lazy_import % conda smithy recipe-lint .
Traceback (most recent call last):
  File "/Users/username/opt/anaconda3/bin/conda-smithy", line 10, in <module>
    sys.exit(main())
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/conda_smithy/cli.py", line 613, in main
    args.subcommand_func(args)
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/conda_smithy/cli.py", line 446, in __call__
    lints, hints = lint_recipe.main(
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/conda_smithy/lint_recipe.py", line 820, in main
    results, hints = lintify(meta, recipe_dir, conda_forge)
  File "/Users/username/opt/anaconda3/lib/python3.8/site-packages/conda_smithy/lint_recipe.py", line 202, in lintify
    license = about_section.get("license", "").lower()
AttributeError: 'CommentedSeq' object has no attribute 'lower'
```

File: recipes/lazy_import/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  description: |
+    `lazy_import` provides a set of functions that load modules, and related attributes, in a lazy fashion.
+    This allows deferring of `ImportErrors` to actual module use-time. Likewise, actual module
+    initialization only takes place at use-time. This is useful when using optional
+    dependencies with heavy loading times and/or footprints, since that cost is
+    only paid if the module is actually used.
+
+    For minimal impact to other code running in the same session `lazy_import`
+    functionality is implemented without the use of import hooks.
+  license:
```
Comment (ocefpaf) [inline recipes/lazy_import/meta.yaml:49]: I don't think that yaml lists work here :-(

You'll need to do :

```yaml
license: GPL-3.0-only and ZPL-2.0
```

That should fix your linting error.

Comment (BastianZim) [reply recipes/lazy_import/meta.yaml:49]: Thank you, will try that!

Comment (BastianZim) [reply recipes/lazy_import/meta.yaml]: Fixed it, thank you!

**Comment by BastianZim** [comment]
The tests fail for windows because the package is using `os.fork()` which is not available there. I'd assume that excludes it from being noarch, correct? The PyPI page says that it's platform-independent though so I just wanted to double-check.

https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python does say that noarch is about the package itself and not its dependencies but the dependency is `os`, not third-party, so not sure if that still applies.

**Comment by BastianZim** [comment]
@ocefpaf Would you mind having a quick look at my comment above, as the package should otherwise be ready for review, thanks!

**Comment by ocefpaf** [comment]
> The tests fail for windows because the package is using `os.fork()` which is not available there. I'd assume that excludes it from being noarch, correct? The PyPI page says that it's platform-independent though so I just wanted to double-check.

It can be noarch if the use of `os.fork` is optional inside the code. Or, and I'm not a Windows expert, if there us any special case where os.fork would be available on Win.

@isuruf what do you think is the best approach here?

**Comment by BastianZim** [comment]
Just to add and apologies for not clarifying this earlier. It seems like this is a problem inside the test function, not the package itself. At least that's what I could find with the error. So I probably jumped to conclusions with this error and it could depend on something entirely different. But I have to admit that this is above my python knowledge so any help is appreciated 😄

**Comment by ocefpaf** [comment]
> Just to add and apologies for not clarifying this earlier. It seems like this is a problem inside the test function, not the package itself. At least that's what I could find with googling the error.
> So I probably jumped to conclusions with this and it could depend on something entirely different. But I have to admit that this is above my level of python knowledge so any help is appreciated.

We may need to skip that test on Windows then.

**Comment by BastianZim** [comment]
Ok, let me move this back to drafts and test a couple of things to narrow it down.

**Comment by BastianZim** [comment]
@ocefpaf I switched the tests to python commands and they're now being run successfully, except for windows where the test skips those sections by itself. 
One of the reasons is `os.fork()`, as mentioned above, but it's due to pytest and not the package so I'd say that it is negligible.

**Comment by BastianZim** [comment]
Thank you!

**Pull request merged**